src/Framework/TestCase.php

```diff
@@ -1133,19 +1133,28 @@ protected function runTest()
             );
         }
 
-        try {
-            $class  = new ReflectionClass($this);
-            $method = $class->getMethod($this->name);
-        } catch (ReflectionException $e) {
-            $this->fail($e->getMessage());
-        }
-
         $testArguments = \array_merge($this->data, $this->dependencyInput);
 
         $this->registerMockObjectsFromTestArguments($testArguments);
 
         try {
-            $testResult = $method->invokeArgs($this, $testArguments);
+            // only use reflection to invoke the method if we have to
+            if ($testArguments) {
```

Comment:
What about using the argument unpacking to remove the reflection altogether?

Reply:
Definitely worth exploring. @jwage Do you want to look into this and update the pull request accordingly?

Reply:
Does not work, results in `Cannot unpack array with string keys` error.

Reply:
Hmkay, using `...array_values($testArguments)` instead of simply `...$testArguments` seems to work.

Reply:
I will test this too and see if it gives us the same level of performance. even when $testArguments is an empty array.